klayout_package/python/kqcircuits/simulations/simulation.py

```diff
@@ -1455,3 +1457,74 @@ def visualise_region(
         else:
             for i, p in enumerate(points):
                 self.cell.shapes(visualisation_layer).insert(pya.DText(f"{label}_{i+1}", p.x, p.y))
+
+    def visualise_ports(self, edge_port_thickness=500):
+        """Visualise all ports in a dedicated 'simulation_ports' layer using get_port_data().
+
+        Args:
+            edge_port_thickness (float): Extra outward extension (µm) for EdgePorts to make them more visible.
+        """
+        dbu = self.layout.dbu
+        port_json = self.get_port_data()
+
+        for port in self.ports:
+        # Try to find a matching entry in port_json
+        port_data_list = [p for p in port_json if p.get("number") == port.number]
+        if not port_data_list:
+            logging.warning(f"Port {port.number} not found in get_port_data() output, skipping visualisation.")
+            continue
+        port_data = port_data_list[0]
+
+        # Label based on port type
+        if isinstance(port, EdgePort):
+            label = f"edge_port_{port.number}"
+        elif isinstance(port, InternalPort):
+            label = f"internal_port_{port.number}"
+        else:
+            logging.warning(f"Unsupported port type {type(port).__name__} for port {port.number}")
+            continue
+
+        # Construct 2D polygon from port_data["polygon"] (ignoring z coordinate)
+        if "polygon" in port_data and port_data["polygon"]:
+            points_2d = [pya.DPoint(p[0], p[1]) for p in port_data["polygon"]]
+
+            if isinstance(port, EdgePort) and edge_port_thickness > 0:
+                    direction = None
```

Comment:
```suggestion
            if isinstance(port, EdgePort) and edge_port_thickness > 0:
                direction = None
```
The code now I think is runnable (can't check because I'm running out of time for today) but indentation can still be improved for style. Try running `black -l 120 -t py310 klayout_package/python/kqcircuits/simulations/simulation.py` and it will fix the style automatically.

Also if you're unsure that the git tool pushed the changes you made correctly, refer to https://github.com/iqm-finland/KQCircuits/pull/118/files to see what is the code that I see.